README.md
# UI Components

A Flutter Package Providing Templates of attractive Login UI's, built with Flutter, ready to be used in your Applications!
No External Packages have been used, so in case, the widgets don't go according to your choice, you can copy-paste the file by checking out the repository on GitHub!

###### Last Update: Added LoginScreen1

## Login Screens

### Login Screen 1
![alt text](https://github.com/RivaanRanawat/flutter_ui_components/blob/master/assets/images/loginScreen1.jpeg?raw=true)

#### Usage
```
LoginScreen1(
    avatarImage: "path/to/your/image.png",
    emailController: _emailController,
    passwordController: _passwordController,
    navigatePage: () {
        // Works when user clicks the sign up button
    },
    onAnotherAccountClick: () {
        // Works when user clicks Another Account Button
    },
    onLoginClick: () {
        // Works when user clicks the Log in Button
    },
)
```

### Login Screen 2
![alt text](https://github.com/RivaanRanawat/flutter_ui_components/blob/master/assets/images/loginScreen2.jpeg?raw=true)

#### Usage
```
LoginScreen2(
    emailController: _emailController,
    passwordController: _passwordController,
    googleLogin: () {
        // Works when user clicks Google Sign In Button
    },
    navigatePage: () {
        // Works when user clicks the sign up button
    },
    onLoginClick: () {
        // Works when user clicks the Log in Button
    },
)
```

### Login Screen 3
![alt text](https://github.com/RivaanRanawat/flutter_ui_components/blob/master/assets/images/loginScreen3.jpeg?raw=true)

#### Usage
```
LoginScreen3(
    emailController: _emailController,
    passwordController: _passwordController,
    navigatePage: () {
        Works when user clicks the sign up button
    },
    onLoginClick: () {
        // Works when user clicks the Log in Button
    },
),
```

## Contribution
Feel Free to Contribute by sending in PR's.


CHANGELOG.md
## [0.0.1] - 28/03/21
* Added Login Screen 1

## [0.0.2] - 28/03/21
* Fixed Login Screen 1

## [0.0.3] - 28/03/21
* Added Login Screen 2

## [0.0.4] - 29/03/21
* Added Login Screen 3


lib/login_screen_1.dart
import 'package:flutter/material.dart';

class LoginScreen1 extends StatefulWidget {
  final TextEditingController emailController;
  final TextEditingController passwordController;
  final String avatarImage;
  final Function onLoginClick;
  final Function googleSignIn;
  final Function navigatePage;
  LoginScreen1({
    @required this.emailController,
    @required this.passwordController,
    @required this.avatarImage,
    @required this.onLoginClick,
    @required this.googleSignIn,
    @required this.navigatePage,
  });
  @override
  _LoginScreen1State createState() => _LoginScreen1State();
}

class _LoginScreen1State extends State<LoginScreen1> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      resizeToAvoidBottomInset: false,
      body: Stack(
        children: [
          Container(
            width: double.infinity,
            height: MediaQuery.of(context).size.height,
            decoration: BoxDecoration(
              image: DecorationImage(
                  image: AssetImage('assets/images/signin_page_background.png'),
                  fit: BoxFit.fill),
            ),
          ),
          Container(
            margin: EdgeInsets.only(top: 120),
            decoration: BoxDecoration(
              color: Color.fromRGBO(53, 55, 88, 1),
              borderRadius: BorderRadius.only(
                topLeft: Radius.circular(37.5),
                topRight: Radius.circular(37.5),
              ),
            ),
            child: Container(
              margin: EdgeInsets.only(top: 90),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.start,
                children: [
                  Text(
                    "Welcome Back",
                    style: TextStyle(
                        fontSize: 17,
                        color: Color.fromRGBO(147, 148, 184, 1),
                        fontWeight: FontWeight.bold),
                  ),
                  SizedBox(height: 30),
                  Container(
                    margin: EdgeInsets.only(top: 22.5, right: 22.5, left: 22.5),
                    child: TextField(
                      controller: widget.emailController,
                      cursorColor: Colors.white,
                      decoration: InputDecoration(
                        enabledBorder: UnderlineInputBorder(
                            borderSide: BorderSide(
                                color: Color.fromRGBO(90, 90, 90, 1))),
                        icon: Icon(
                          Icons.email,
                          color: Colors.white,
                        ),
                        contentPadding: EdgeInsets.all(11.25),
                        hintText: "Email",
                        hintStyle: TextStyle(
                          color: Colors.white,
                        ),
                      ),
                    ),
                  ),
                  Container(
                    margin: EdgeInsets.only(top: 22.5, right: 22.5, left: 22.5),
                    child: TextField(
                      controller: widget.passwordController,
                      cursorColor: Colors.white,
                      decoration: InputDecoration(
                        enabledBorder: UnderlineInputBorder(
                          borderSide: BorderSide(
                            color: Color.fromRGBO(90, 90, 90, 1),
                          ),
                        ),
                        icon: Icon(Icons.lock, color: Colors.white),
                        contentPadding: EdgeInsets.all(11.25),
                        hintText: "Password",
                        hintStyle: TextStyle(
                          color: Colors.white,
                        ),
                      ),
                      obscureText: true,
                    ),
                  ),
                  Container(
                    width: MediaQuery.of(context).size.width,
                    height: 60,
                    margin: EdgeInsets.only(top: 40, left: 30, right: 30),
                    child: ElevatedButton(
                      style: ButtonStyle(
                        backgroundColor: MaterialStateProperty.all<Color>(
                          Color.fromRGBO(255, 87, 34, 1),
                        ),
                        shape:
                            MaterialStateProperty.all<RoundedRectangleBorder>(
                          RoundedRectangleBorder(
                            borderRadius: BorderRadius.circular(30),
                          ),
                        ),
                      ),
                      onPressed: widget.onLoginClick,
                      child: Text(
                        "Log in",
                        style: TextStyle(
                            fontSize: 17, fontWeight: FontWeight.bold),
                      ),
                    ),
                  ),
                  Container(
                    width: MediaQuery.of(context).size.width,
                    height: 60,
                    margin: EdgeInsets.only(top: 20, right: 30, left: 30),
                    child: TextButton(
                      style: ButtonStyle(
                        // backgroundColor: MaterialStateProperty.all<Color>(
                        //   Colors.transparent,
                        // ),
                        shape:
                            MaterialStateProperty.all<RoundedRectangleBorder>(
                          RoundedRectangleBorder(
                            borderRadius: BorderRadius.circular(30),
                          ),
                        ),
                      ),
                      onPressed: widget.googleSignIn,
                      child: Text(
                        "Sign In With Google",
                        style: TextStyle(
                          color: Colors.white,
                          fontSize: 17,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                  ),
                  Expanded(
                    child: Container(
                      margin: EdgeInsets.only(top: 60),
                      width: double.infinity,
                      decoration: BoxDecoration(
                        color: Colors.white,
                        borderRadius: BorderRadius.only(
                          topLeft: Radius.circular(40),
                          topRight: Radius.circular(40),
                        ),
                      ),
                      child: Column(
                        children: [
                          IconButton(
                              icon: Icon(Icons.keyboard_arrow_up),
                              onPressed: widget.navigatePage),
                          Text(
                            "Sign up",
                            style: TextStyle(
                                fontSize: 17, fontWeight: FontWeight.bold),
                          ),
                        ],
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
          Container(
            margin: EdgeInsets.only(
              left: MediaQuery.of(context).size.width / 2 - 50,
              top: MediaQuery.of(context).size.height / 10.1,
            ),
            child: CircleAvatar(
              radius: 50,
              backgroundImage: AssetImage(widget.avatarImage),
            ),
          ),
        ],
      ),
    );
  }
}


lib/login_screen_3.dart
import "package:flutter/material.dart";

class LoginScreen3 extends StatefulWidget {
  final TextEditingController emailController;
  final TextEditingController passwordController;
  final Function onLoginClick;
  final Function navigatePage;
  LoginScreen3({
    @required this.emailController,
    @required this.passwordController,
    @required this.onLoginClick,
    @required this.navigatePage,
  });
  @override
  _LoginScreen3State createState() => _LoginScreen3State();
}

class _LoginScreen3State extends State<LoginScreen3> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: Stack(
          children: [
            Container(
              height: MediaQuery.of(context).size.height,
              width: MediaQuery.of(context).size.width,
              decoration: BoxDecoration(
                image: DecorationImage(
                  image: AssetImage(
                      "assets/images/background_image_one_signin.png"),
                  fit: BoxFit.fill,
                ),
              ),
            ),
            Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Container(
                  margin: EdgeInsets.only(left: 40, right: 8, top: 155),
                  child: Text(
                    "Welcome\nBack",
                    style: TextStyle(
                        color: Colors.white,
                        fontSize: 30,
                        fontWeight: FontWeight.bold),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: Container(
                    margin: EdgeInsets.only(
                        left: 30,
                        right: 30,
                        top: MediaQuery.of(context).size.height * 0.24),
                    width: double.infinity,
                    child: TextField(
                      controller: widget.emailController,
                      style: TextStyle(color: Color.fromRGBO(41, 41, 41, 1)),
                      decoration: InputDecoration(
                        hintText: "Email",
                        hintStyle:
                            TextStyle(color: Color.fromRGBO(121, 121, 121, 1)),
                      ),
                      keyboardType: TextInputType.emailAddress,
                    ),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: Container(
                    margin: EdgeInsets.only(left: 30, right: 30),
                    width: double.infinity,
                    child: TextField(
                      controller: widget.passwordController,
                      style: TextStyle(color: Color.fromRGBO(41, 41, 41, 1)),
                      decoration: InputDecoration(
                        hintText: "Password",
                        hintStyle:
                            TextStyle(color: Color.fromRGBO(121, 121, 121, 1)),
                      ),
                      keyboardType: TextInputType.emailAddress,
                      obscureText: true,
                    ),
                  ),
                ),
                Row(
                  children: [
                    Padding(
                      padding: const EdgeInsets.all(50),
                      child: Text(
                        "Sign in",
                        style: TextStyle(
                          color: Colors.black,
                          fontSize: 25,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                    Container(
                        margin: EdgeInsets.only(left: 120),
                        child: ClipOval(
                          child: Material(
                            color: Color.fromRGBO(78, 82, 91, 1),
                            child: InkWell(
                              child: SizedBox(
                                width: 56,
                                height: 56,
                                child: Icon(
                                  Icons.arrow_right,
                                  color: Colors.white,
                                ),
                              ),
                              onTap: widget.onLoginClick,
                            ),
                          ),
                        ))
                  ],
                ),
                SizedBox(height: MediaQuery.of(context).size.height * 0.014),
                Padding(
                  padding: const EdgeInsets.only(left: 40.0),
                  child: TextButton(
                    child: Text(
                      "Sign up",
                      style: TextStyle(
                        color: Colors.black,
                        fontWeight: FontWeight.bold,
                        decoration: TextDecoration.underline,
                        fontSize: 15,
                      ),
                    ),
                    onPressed: widget.navigatePage,
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}


lib/login_screen_2.dart
import 'package:flutter/material.dart';

class LoginScreen2 extends StatefulWidget {
  final TextEditingController emailController;
  final TextEditingController passwordController;
  final Function onLoginClick;
  final Function googleLogin;
  final Function navigatePage;

  LoginScreen2({
    @required this.emailController,
    @required this.passwordController,
    @required this.onLoginClick,
    @required this.googleLogin,
    @required this.navigatePage
  });
  @override
  _LoginScreen2State createState() => _LoginScreen2State();
}

class _LoginScreen2State extends State<LoginScreen2> {
  FocusNode myFocusNode;
  final Color primaryColor = Color(0xff18203d);
  final Color secondaryColor = Color(0xff232c51);
  final Color logoGreen = Color(0xff25bcbb);

  @override
  void initState() {
    super.initState();
    myFocusNode = FocusNode();
  }

  @override
  void dispose() {
    myFocusNode.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: primaryColor,
      appBar: AppBar(
        backgroundColor: Colors.transparent,
        elevation: 0,
      ),
      body: Container(
        alignment: Alignment.topCenter,
        margin: EdgeInsets.symmetric(horizontal: 30),
        child: SingleChildScrollView(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: <Widget>[
              Text(
                'Login To Continue!',
                textAlign: TextAlign.center,
                style: TextStyle(color: Colors.white, fontSize: 28),
              ),
              SizedBox(height: 20),
              Text(
                'Enter your email and password below to continue!',
                textAlign: TextAlign.center,
                style: TextStyle(color: Colors.white, fontSize: 14),
              ),
              SizedBox(
                height: 50,
              ),
              Container(
                padding: EdgeInsets.symmetric(horizontal: 10, vertical: 5),
                decoration: BoxDecoration(
                    color: secondaryColor,
                    border: Border.all(color: Colors.blue)),
                child: TextFormField(
                  controller: widget.emailController,
                  style: TextStyle(color: Colors.white),
                  decoration: InputDecoration(
                    contentPadding: EdgeInsets.symmetric(horizontal: 10),
                    labelText: "Email",
                    labelStyle: TextStyle(color: Colors.white),
                    icon: Icon(
                      Icons.email,
                      color: Colors.white,
                    ),
                    border: InputBorder.none,
                  ),
                  keyboardType: TextInputType.emailAddress,
                  textInputAction: TextInputAction.next,
                  onFieldSubmitted: (_) => myFocusNode.requestFocus(),
                ),
              ),
              SizedBox(height: 20),
              Container(
                padding: EdgeInsets.symmetric(horizontal: 10, vertical: 5),
                decoration: BoxDecoration(
                    color: secondaryColor,
                    border: Border.all(color: Colors.blue)),
                child: TextFormField(
                  focusNode: myFocusNode,
                  controller: widget.passwordController,
                  style: TextStyle(color: Colors.white),
                  decoration: InputDecoration(
                    contentPadding: EdgeInsets.symmetric(horizontal: 10),
                    labelText: "Password",
                    labelStyle: TextStyle(color: Colors.white),
                    icon: Icon(
                      Icons.lock,
                      color: Colors.white,
                    ),
                    border: InputBorder.none,
                  ),
                  obscureText: true,
                ),
              ),
              SizedBox(height: 30),
              MaterialButton(
                elevation: 0,
                minWidth: double.maxFinite,
                height: 50,
                onPressed: widget.onLoginClick,
                color: logoGreen,
                child: Text('Login',
                    style: TextStyle(color: Colors.white, fontSize: 16)),
                textColor: Colors.white,
              ),
              SizedBox(height: 20),
              MaterialButton(
                elevation: 0,
                minWidth: double.maxFinite,
                height: 50,
                onPressed: widget.googleLogin,
                color: Colors.blue,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    Text('Sign In using Google',
                        style: TextStyle(color: Colors.white, fontSize: 16)),
                  ],
                ),
                textColor: Colors.white,
              ),
              SizedBox(height: 100),
              Align(
                alignment: Alignment.bottomCenter,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: <Widget>[
                    GestureDetector(
                      onTap: widget.navigatePage,
                      child: Text('Sign Up?',
                          textAlign: TextAlign.center,
                          style: TextStyle(
                              color: Colors.white,
                              fontSize: 20,
                              fontWeight: FontWeight.bold)),
                    ),
                  ],
                ),
              )
            ],
          ),
        ),
      ),
    );
  }
}